api/src/main/java/org/apache/iceberg/Accessors.java

```diff
@@ -26,6 +26,20 @@
 import org.apache.iceberg.types.TypeUtil;
 import org.apache.iceberg.types.Types;
 
+/**
+ * Position2Accessor and Position3Accessor here is an optimization. For a nested schema like:
+ *
+ * root
```

Comment:
thank you very much for the update, sorry for the back and forth, just one last thing: the example block here needs to be in `<pre></pre>`, otherwise they will collapse.